EventFiltering/PWGHF/HFFilterHelpers.h

```diff
@@ -610,19 +629,40 @@ inline int8_t HfFilterHelper::isSelectedTrackForSoftPionOrBeauty(const T& track,
   return retValue;
 }
 
-/// Basic selection of proton candidates
+/// Basic selection of proton or deuteron candidates
 /// \param track is a track
 /// \param trackPar is a track parameter
 /// \param activateQA flag to activate the filling of QA histos
 /// \param hProtonTPCPID histo with NsigmaTPC vs. p
 /// \param hProtonTOFPID histo with NsigmaTOF vs. p
 /// \param forceTof flag to force TOF PID
+/// \param trackSpecies flag to choose proton or deuteron
 /// \return true if track passes all cuts
 template <typename T1, typename T2, typename H2>
-inline bool HfFilterHelper::isSelectedProton4Femto(const T1& track, const T2& trackPar, const int& activateQA, H2 hProtonTPCPID, H2 hProtonTOFPID, bool forceTof)
+inline bool HfFilterHelper::isSelectedTrack4Femto(const T1& track, const T2& trackPar, const int& activateQA, H2 hTPCPID, H2 hTOFPID, bool forceTof, const int& trackSpecies)
```

Comment:
I would separate also the argument to force the TOF between proton and deuteron (it might well be that we can avoid it for the deuteron, since the rejection factor should be much larger than the one for the proton - to be tested anyway). You can consider a datamember for this as well, as for the other parameters.

Reply:
good idea!